dash/dash-renderer/src/utils/clientsideFunctions.ts

```diff
@@ -18,6 +18,30 @@ const set_props = (id: string | object, props: {[k: string]: any}) => {
     }
 };
 
+// Clean url code adapted from https://github.com/braintree/sanitize-url/blob/main/src/constants.ts
+// to allow for data protocol.
+const invalidProtocols = /^([^\w]*)(javascript|vbscript)/im;
+const newLines = /&(tab|newline);/gi;
+
+// eslint-disable-next-line no-control-regex
+const ctrlChars = /[\u0000-\u001F\u007F-\u009F\u2000-\u200D\uFEFF]/gim;
+const htmlEntities = /&#(\w+)(^\w|;)?/g;
+
+const clean_url = (url: string, fallback = 'about:blank') => {
+    if (url === '') {
+        return url;
+    }
+    const cleaned = url
```

Comment:
Do we need to include the `decodeHtmlCharacters` step as in `braintree/sanitize-url`?